pom.xml

```diff
@@ -44,7 +44,7 @@
         <url>https://github.com/apache/skywalking</url>
         <connection>scm:git:https://github.com/apache/skywalking.git</connection>
         <developerConnection>scm:git:https://github.com/apache/skywalking.git</developerConnection>
-        <tag>v9.4.0</tag>
+        <tag>HEAD</tag>
```

Comment:
See #10512 did this(to specific tag) automatically. Wonder what we should put here.

Reply:
> See #10512 did this(to specific tag) automatically. Wonder what we should put here.

So #10512 accidentally submit this, here it should be `HEAD` and it will only be changed when doing a release, but should not in master branch

Reply:
> So https://github.com/apache/skywalking/pull/10512 accidentally submit this, here it should be HEAD and it will only be changed when doing a release, but should not be in the master branch

But every time, when I do the release scripts, it is changed automatically. And then, we merge that release branch into the master branch. 
What could we do to avoid this?

Reply:
Have you tried that? This is what I did before, but it is not changed back. As I mentioned, I never manually change this, but from the commit log, it is from my 9.4 release version bumping up.

Reply:
> Have you tried that? This is what I did before, but it is not changed back. 


I tried that. 


> As I mentioned, I never manually change this, but from the commit log, it is from my 9.4 release version bumping up.

It's from 9.4 release, that's the problem. If it's always like that, the main branch should be 9.5.0(previous version) now, but it's 9.4.0, and I assume it will be 9.4.0 forever even when you release 9.6.0. This is the issue, if it is not HEAD, it assumes you have already set the tag manually and it won't change it, I didn't verify this point though.

Reply:
You can simply check all previous releases tag and compare the  commits before and after that tag, you can easily find how the tag is modified

Reply:
Meanwhile, I want to say, I did `9.3.0` release, #10077, with the same script, it was correct. 

<img width="1484" alt="image" src="https://github.com/apache/skywalking/assets/5441976/8c31c2f4-2a50-482c-a42e-dac9008cf87d">

But the version of apache parent was changed from 25 to 29.
```
<parent>
        <groupId>org.apache</groupId>
        <artifactId>apache</artifactId>
        <version>29</version>
        <relativePath />
    </parent>
```

So, I am wondering whether this is an existing bug, and we may need to fix this as well.

Reply:
@kezhenxu94 Please check this, https://github.com/apache/skywalking/pull/11209